profiling/src/capi.rs

```diff
@@ -3,6 +3,62 @@
 
 use crate::bindings::zend_execute_data;
 use crate::RUNTIME_ID;
+use std::borrow::Cow;
+use std::fmt::{Display, Formatter};
+use std::marker::PhantomData;
+
+/// A non-owning, not necessarily null terminated, not utf-8 encoded, borrowed
+/// string. Must satisfy the requirements of [core::slice::from_raw_parts],
+/// notably it must not use the nul pointer even when the length is 0.
```

Comment:
I'd like to note that this constraint (not-NULL) doesn't hold for some usages for zai_string_view on the C side. For everything passed to the profiler, this currently holds though.

Reply:
The `components/string_view/string_view.h` version notes this requirement, maybe I should use that instead? They have the same layout.